src/ephemeris/set_library_permissions.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+'''Tool to set permissions for all datasets of a given Galaxy Data Library'''
+
+import argparse
+import logging as log
+from bioblend import galaxy
+import sys, time, os
+
+from .common_parser import get_common_args
+# Print iterations progress
+def printProgressBar (iteration, total, prefix = '', suffix = '', decimals = 1, length = 100, fill = '█', printEnd = "\r"):
+    """
+    Call in a loop to create terminal progress bar
+    @params:
+        iteration   - Required  : current iteration (Int)
+        total       - Required  : total iterations (Int)
+        prefix      - Optional  : prefix string (Str)
+        suffix      - Optional  : suffix string (Str)
+        decimals    - Optional  : positive number of decimals in percent complete (Int)
+        length      - Optional  : character length of bar (Int)
+        fill        - Optional  : bar fill character (Str)
+        printEnd    - Optional  : end character (e.g. "\r", "\r\n") (Str)
+    """
+    percent = ("{0:." + str(decimals) + "f}").format(100 * (iteration / float(total)))
+    filledLength = int(length * iteration // total)
+    bar = fill * filledLength + '-' * (length - filledLength)
+    print(f'\r{prefix} |{bar}| {percent}% {suffix}', end = printEnd)
+    # Print New Line on Complete
+    if iteration == total: 
+        print()
+
+def get_datasets(gi, library_id) -> [str]:
+    objects = gi.libraries.show_dataset(library_id=library_id, dataset_id='')
+    datasets = []
+    for index in range(len(objects)):
+       if objects[index]['type'] == 'file':
+           datasets.append(objects[index]['id'])
+    if datasets == []:
+        sys.exit("No datasets in library!")
+    else:
+        return datasets
+
+def set_permissions(gi, library_id, role_ids):
+    log.info("Your library_id is " + library_id + "\n")
+    role_msg = "Your roles are: " + " ".join(role_ids)
+    log.info(role_msg)
```

Comment:
this way the string will only get properly templated *if* you're above the correct log level.
```suggestion
    log.info("Your roles are: %s", " ".join(role_ids))
```